BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +57,502 @@ public struct AttributedLabel: Element, Hashable {
+        /// The touch handling logic explicitly tracks the active links when touches begin, so if you drag outside
+        /// the link and touch up over another link, it just cancels the tap rather than accidentally opening
+        /// a different link.
+        private var trackingLinks: [Link]?
+
+        private var linkDetectionTypes: Set<LinkDetectionType> = []
+        private var linkAttributes: [NSAttributedString.Key: Any] = [:]
+        private var activeLinkAttributes: [NSAttributedString.Key: Any] = [:]
+
+        private var links: [Link] = [] {
+            didSet {
+                isUserInteractionEnabled = !links.isEmpty
+            }
+        }
+
+        private var textRectOffset: UIOffset = .zero {
             didSet {
                 if oldValue != textRectOffset {
                     setNeedsDisplay()
                 }
             }
         }
 
+        /// The view needs to keep track of the current index for the accessibility rotor.
+        private var accessibilityLinkIndex = -1
+
+        /// These elements need to be retained by the view, and cannot be created inside the
+        /// `accessibilityCustomRotors` getter.
+        private var accessibilityLinks: [LinkAccessibilityElement] = []
+
+        override var accessibilityCustomRotors: [UIAccessibilityCustomRotor]? {
+            set { fatalError() }
+            get {
+                [
+                    UIAccessibilityCustomRotor(systemType: .link) { [weak self] predicate in
+                        guard let self = self else {
+                            return nil
+                        }
+
+                        self.accessibilityLinkIndex += predicate.searchDirection == .next ? 1 : -1
+                        self.accessibilityLinkIndex = min(self.accessibilityLinks.count - 1, self.accessibilityLinkIndex)
+                        self.accessibilityLinkIndex = max(0, self.accessibilityLinkIndex)
+
+                        let link = self.accessibilityLinks[self.accessibilityLinkIndex]
+                        return UIAccessibilityCustomRotorItemResult(targetElement: link, targetRange: nil)
+                    },
+                ]
+            }
+        }
+
+        var linkHandler: LinkHandler?
+
+        func update(model: AttributedLabel, linkHandler: LinkHandler?, isMeasuring: Bool) {
+            let previousString = attributedText?.string
+
+            linkAttributes = model.linkAttributes
+            activeLinkAttributes = model.activeLinkAttributes
+            linkDetectionTypes = model.linkDetectionTypes ?? []
+            attributedText = model.attributedText
+                .applyingDefaultFont()
+                .replacingLinkAttributes()
+            numberOfLines = model.numberOfLines
+            textRectOffset = model.textRectOffset
+            isAccessibilityElement = model.isAccessibilityElement
+            self.linkHandler = linkHandler
+
+            if !isMeasuring, previousString != attributedText?.string {
```

Comment:
`previousString` stores the string before we update `attributedText`, then we update it from the `model` before this comparison; so this is just checking to see if the string actually changed. Realizing this is the incorrect thing to do though, since we're checking for links in the attributes. Need to update.

Reply:
- [ ] Fix this check